python/ic_azure/azure_discover_dimensions.py

```diff
@@ -4,71 +4,100 @@
+                interval="PT1H",
+                metricnames=metric,
+                aggregation="Total",
+                result_type="Metadata",
+                filter=queryfilter,
+                metricnamespace=metric_namespace
+            )
+        except Exception as error:
+            sys.exit(f"Client request failed. {error}")
+
+        return metrics_data
+
+    # Method to retrieve dimensions from Azure's resource
+    def get_dimensions(self, resource, metric, dimension, extent, metric_namespace):
+
+        # Declare variables
+        dimensionsList = []
+
         # Read resource from config using key
         if not resource.startswith("/subscriptions"):
             resource = self.resources.get(resource)
-
-        # Retrieve instance data
-        metrics_data = self._client.metrics.list(
-            resource,
-            timespan="{}/{}".format(
-                start_time.strftime('%Y-%m-%dT%H:%M:%SZ'),
-                end_time.strftime('%Y-%m-%dT%H:%M:%SZ')
-            ),
-            interval="PT1H",
-            metricnames=metric,
-            aggregation="Total",
-            result_type="Metadata",
-            filter=dimension + " eq '*'",
-            metricnamespace=metric_namespace
-        )
+        
+        filter = dimension + " eq '*'"
+        result = self.get_data(resource, metric, filter, metric_namespace)
 
         # Loop through metric data and retrieve instances
-        for item in metrics_data.value:
+        for item in result.value:
             for timeserie in item.timeseries:
                 for data in timeserie.metadatavalues:
-
-                    # Don't add duplicates into roles list
-                    if data.__dict__.get("value") in rolesList:
+                    # Don't add duplicates into dimensions list
+                    name = data.__dict__.get("value")
+                    if name in dimensionsList:
                         continue
-
-                    rolesList.append(data.__dict__.get("value"))
-
-        return rolesList
-
+                    
+                    dimensionsList.append({"{#DIMENSION}": name, "{#ROLE_NAME}": name})
+                    
+                    # Get second dimensions (extents) for the discovered resource
+                    if extent:
+                        extent_filter = f"{dimension} eq '{name}' and {extent} eq '*'"
+                        extent_data = self.get_data(resource, metric, extent_filter, metric_namespace)
+                        
+                        # Loop through metric data and retrieve instances
+                        for property in extent_data.value:
+                            for serie in property.timeseries:
+                                for info in serie.metadatavalues:
+                                    # Don't add duplicates into dimensions list
+                                    name2 = info.__dict__.get("value")
+                                    if name2 in dimensionsList:
```

Comment:
This duplicate check won't work either. Also, this duplicate check should probably be "first dimension" specific because there may be multiple similar devices on multiple nodes and they should all be added to `dimensionsList`.